src/shared/components/oncoprint/ResultsViewOncoprint.spec.tsx

```diff
@@ -0,0 +1,112 @@
+        const oncoprintView = initResultsViewWithSortByParam({sortByParam: 'case_id', columnMode: 'sample'});
+        assert.deepEqual( oncoprintView.sortConfig.order, ["Sample1_key","Sample2_key","Sample3_key"]);
+    });
+
+    it('`case_id` provides sorted patient config to oncoprint', () => {
+        const oncoprintView = initResultsViewWithSortByParam({sortByParam:'case_id', columnMode:'patient'});
+        assert.deepEqual( oncoprintView.sortConfig.order, ["Patient1_key","Patient2_key","Patient3_key"]);
+    });
+
+    it('`case_list` provides sorted sample config to oncoprint when case list is available', () => {
+        const oncoprintView = initResultsViewWithSortByParam({sortByParam: 'case_list', columnMode: 'sample', caselistEnabled: true});
+        assert.deepEqual(oncoprintView.sortConfig.order,  ["Sample2_key","Sample3_key","Sample1_key"]);
+    });
+
+    it('`case_list` provides sorted patient config to oncoprint when case list is available', () => {
+        const oncoprintView = initResultsViewWithSortByParam({sortByParam: 'case_list', columnMode: 'patient', caselistEnabled: true});
+        assert.deepEqual(oncoprintView.sortConfig.order,  ["Patient2_key","Patient3_key","Patient1_key"]);
+    });
+
+    it('`case_list` provides no sort config to oncoprint when case list is unavailable', () => {
+        const oncoprintView = initResultsViewWithSortByParam({sortByParam: 'case_list', columnMode: 'patient', caselistEnabled: false});
+        assert.isUndefined(oncoprintView.sortConfig.order);
+    });
+
+    it('`data` provides data-based sort config to oncoprint', () => {
+        const oncoprintView = initResultsViewWithSortByParam({sortByParam: 'data'});
+        assert.isUndefined(oncoprintView.sortConfig.order);
+        assert.isFalse(oncoprintView.sortByMutationType);
+        assert.isFalse(oncoprintView.sortByDrivers);
+    });
+
+    it('`data_mut` provides data+mutation-based sort config to oncoprint', () => {
+        const oncoprintView = initResultsViewWithSortByParam({sortByParam: 'data_mut'});
+        assert.isUndefined(oncoprintView.sortConfig.order);
+        assert.isTrue(oncoprintView.sortByMutationType);
+        assert.isFalse(oncoprintView.sortByDrivers);
+    });
+
+    it('`data_dri` provides data+driver-based sort config to oncoprint', () => {
+        const oncoprintView = initResultsViewWithSortByParam({sortByParam: 'data_dri'});
+        assert.isUndefined(oncoprintView.sortConfig.order);
+        assert.isFalse(oncoprintView.sortByMutationType);
+        assert.isTrue(oncoprintView.sortByDrivers);
+    });
+
+    it('`data_mut_dri` provides data+mutation+driver-based sort config to oncoprint', () => {
+        const oncoprintView = initResultsViewWithSortByParam({sortByParam: 'data_mut_dri'});
+        assert.isUndefined(oncoprintView.sortConfig.order);
+        assert.isTrue(oncoprintView.sortByMutationType);
+        assert.isTrue(oncoprintView.sortByDrivers);
+    });
+
```

Comment:
tests look great

Reply:
Thnx, I am getting the hang of JS test :)